### ISSUE TITLE
[ISSUE] Provider bug: terraform replaces workspace due to deployment_name prefix

### ISSUE THREAD

**Issue by hamza-tumturk** [databrickslabs/terraform-provider-databricks#382]
Hi,

when creating a workspace in our account, the deployment name is getting prefixed with our account prefix, which is intended behaviour. Creating the resource with terraform, we don't additionally set the prefix, as it automatically gets set by databricks. So the deployment name will eventually end up being `<prefix>-<deployment_name>`.
The issue starts when rerunning the terraform deployment, as terraform will not be aware of the prefixed deployment name and tries to recreate the resource.
As a workaround, I tried to prefix the resource myself, rather than letting databricks do it, but the resource then I end up getting a deployment with double prefix like`<prefix>-<prefix>-<deployment_name>` and the resource still recreates on a terraform rerun.

### Terraform Version
`Terraform v0.12.25`

### Affected Resource(s)
- databricks_mws_workspaces

### Environment variable names
`AWS`

### Terraform Configuration Files
```hcl
resource "databricks_mws_workspaces" "this" {
  provider        = databricks.mws
  account_id      = var.databricks_account_id
  aws_region      = var.aws_region
  workspace_name  = local.workspace_name
  deployment_name = local.deployment_name

  credentials_id            = databricks_mws_credentials.this.credentials_id
  storage_configuration_id  = databricks_mws_storage_configurations.this.storage_configuration_id
  network_id                = databricks_mws_networks.this.network_id
  verify_workspace_runnning = true

}
```
which correspond in the UI as
![image](https://user-images.githubusercontent.com/19170248/96969050-c477b300-1509-11eb-93e5-d06cf176e23f.png)
### Debug Output

```
  # databricks_mws_workspaces.this must be replaced
-/+ resource "databricks_mws_workspaces" "this" {
        account_id                = (sensitive value)
        aws_region                = "us-east-1"
      ~ creation_time             = 1603405747119 -> (known after apply)
        credentials_id            = "XXX"
      ~ deployment_name           = "prefix-deployment_name" -> "deployment_name" # forces replacement
```

### Expected Behavior
Terraform shouldn't recreate the workspace resource for accounts when databricks adds a prefix, but rather detect the prefix and consider it in the change plan.

### Actual Behavior
Terraform recreates the workspace resource.

### Steps to Reproduce
1. `terraform apply`

**Comment by hamza-tumturk** [comment]
In the meantime, a possible workaround for this issue is just to add a lifecycle to ignore the changes on the `deployment_name` for the resource `databricks_mws_workspaces`
```
lifecycle {
  ignore_changes = [
    deployment_name
  ]
}
```

**Comment by nfx** [comment]
Fixed in #400
